src/stream/src/executor/gap_fill.rs

```diff
@@ -0,0 +1,1856 @@
+        let Self {
+            mut managed_state,
+            mut cache_manager,
+            schema,
+            chunk_size,
+            time_column_index,
+            fill_columns,
+            gap_interval,
+            ctx,
+            input,
+        } = *self;
+
+        let mut input = input.execute();
+
+        let barrier = expect_first_barrier(&mut input).await?;
+        let first_epoch = barrier.epoch;
+        yield Message::Barrier(barrier);
+        managed_state.init_epoch(first_epoch).await?;
+
+        // Start with an empty cache - use lazy loading strategy.
+        // Data will be loaded on-demand when gap filling operations require it.
+
+        // Calculate and validate gap interval once at initialization
+        let dummy_row = OwnedRow::new(vec![]);
+        let interval_datum = gap_interval.eval_row_infallible(&dummy_row).await;
+        let interval = interval_datum
+            .ok_or_else(|| anyhow::anyhow!("Gap interval expression returned null"))?
+            .into_interval();
+
+        // Validate that gap interval is not zero
+        if interval.months() == 0 && interval.days() == 0 && interval.usecs() == 0 {
+            Err(anyhow::anyhow!("Gap interval cannot be zero"))?;
+        }
+
+        #[for_await]
+        for msg in input {
+            match msg? {
+                Message::Chunk(chunk) => {
+                    let chunk = chunk.compact();
+                    let mut output_rows = Vec::new();
+
+                    for (op, row_ref) in chunk.rows() {
+                        let row = row_ref.into_owned_row();
+
+                        match op {
+                            Op::Insert | Op::UpdateInsert => {
+                                let cache_key = managed_state.serialize_time_to_cache_key(&row);
+
+                                // Find previous and next original row neighbors.
+                                let prev_original = cache_manager
+                                    .find_robust_prev_original(&cache_key, &managed_state)
+                                    .await?;
+                                let next_original = cache_manager
+                                    .find_robust_next_original(&cache_key, &managed_state)
+                                    .await?;
+
+                                // If both neighbors exist, delete fill rows between them.
+                                if let (Some((prev_key, _)), Some((next_key, _))) =
+                                    (&prev_original, &next_original)
+                                {
+                                    let filled_rows_to_delete = cache_manager
+                                        .scan_filled_rows_between(
+                                            prev_key,
+                                            next_key,
+                                            &managed_state,
+                                        )
+                                        .await?;
```

Comment:
I'm not sure why we need to always delete all rows. For `null` / `locf`, we can simply replace `inserted_row <-> gap_filled row` in-place? Only for `interpolate` we need to delete all rows. Is this meant for future optimization?